Session.js

```diff
@@ -49,21 +49,33 @@ export default class Session {
     await this.refresh();
   }
 
+  /**
+   * Registers a handler that is executed when an event listener fires.
+   *
+   * @param {string} eventType - An event such as change.
+   * @param {function} handler - A handler function called when
+   *   an event occurs.
+   *
+   * @returns {function} A unique function to remover the listener.
+   */ 
   on(eventType, handler) {
     if(typeof eventType !== 'string') {
       throw new TypeError('"eventType" must be a string.');
     }
     if(typeof handler !== 'function') {
       throw new TypeError('"handler" must be a function.');
     }
-
+    // this should return a Map for the eventType.
     const listeners = this._eventTypeListeners.get(eventType);
     if(!listeners) {
       throw new Error(`Event "${eventType}" is not supported.`);
     }
 
     const remover = () => listeners.delete(remover);
+    // use the function as a unique key in the event listener Map.
     listeners.set(remover, handler);
```

Comment:
The implementation here is really weird to me (and, apparently, I'm the one that wrote it!). Why not, instead of using a Map of Maps for `_eventTypeListeners` ... use a Map of Sets? Sets preserve insertion order -- and there is no reason for this special unique key AFAICT. So, if we change this to use a Set internally, the remove function just removes the handler from the Set. I wonder if this was done because of some compatibility concern that we no longer have. I recommend we change it.

Reply:
that actually sounds good to me. Let me test it first actually and make sure that Set does remove and update correctly. The only issues is every project that depends on it might break: https://github.com/search?utf8=%E2%9C%93&q=in%3Afile+bedrock-web-session&type=Code

but it's not that big of a deal and I don't think this functionality is even used in most browsers.

Reply:
@dlongley changes are in this PR: https://github.com/digitalbazaar/bedrock-web-session/pull/4/commits/01bf803870b37ad80076295263fffdc07df5b5f0

can confirm this works in the wallet in chapi.